src/plugins/vis_builder/public/application/components/data_tab/index.tsx

```diff
@@ -3,19 +3,265 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { DropResult, EuiDragDropContext } from '@elastic/eui';
 import { FieldSelector } from './field_selector';
 
 import './index.scss';
 import { ConfigPanel } from './config_panel';
+import { useAggs, useVisualizationType } from '../../utils/use';
+import { useTypedDispatch, useTypedSelector } from '../../utils/state_management';
+
+import {
+  reorderAgg,
+  updateAggConfigParams,
+} from '../../utils/state_management/visualization_slice';
+import { useOpenSearchDashboards } from '../../../../../opensearch_dashboards_react/public';
+import { VisBuilderServices } from '../../../types';
+import { createNewAggConfig } from './utils/get_valid_aggregations';
+import { DropboxDisplay } from './dropbox';
 
 export const DATA_TAB_ID = 'data_tab';
 
+export interface SchemaDisplayStates {
+  [key: string]: DropboxDisplay[];
+}
+
 export const DataTab = () => {
+  // Field Selector panel States
+  const fieldSelectorGroups = [
+    'preDefinedCountMetric',
+    'categoricalFields',
+    'numericalFields',
+    'metaFields',
+  ];
+
+  // Config panel States
+  const vizType = useVisualizationType();
+  const editingState = useTypedSelector(
+    (state) => state.visualization.activeVisualization?.draftAgg
+  );
+  const schemas = vizType.ui.containerConfig.data.schemas;
+  const {
+    services: {
+      data: {
+        search: { aggs: aggService },
+      },
+    },
+  } = useOpenSearchDashboards<VisBuilderServices>();
+
+  const aggProps = useAggs();
+  const [schemaDisplayStates, setSchemaDisplayStates] = useState<SchemaDisplayStates>(() => {
```

Comment:
nit: If my understanding of this state is correct, it is a way to know which fields are active for which chema. Do you think calling it something like `schemaFieldAssignments` or `activeSchemaFields` might make it easier to understand what this is?